go.mod

```diff
@@ -145,3 +155,8 @@ replace (
 	k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.32.3
 	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.32.3
 )
+
+replace (
+	github.com/openshift/client-go => github.com/openshift/client-go v0.0.0-20241107164952-923091dd2b1a
+	k8s.io/kubernetes => k8s.io/kubernetes v1.32.2
```

Comment:
this is probably the biggest open issue about this PR. Depending again on `k8s.io/kubernetes` is something I'm extremely reluctant to because it brings a world of pain for rebases/updates. We spent a fair bit of energy to remove this dep so even though this PR is compelling, is still a debatable proposition.

Reply:
> this is probably the biggest open issue about this PR. Depending again on `k8s.io/kubernetes` is something I'm extremely reluctant to because it brings a world of pain for rebases/updates. We spent a fair bit of energy to remove this dep so even though this PR is compelling, is still a debatable proposition.

We were discussing about that in the initial iterations of this PR.
Depending on `k8s.io/kubernetes` is the price that we pay for updating the default functions automatically.

Reply:
correct. This is the main/only friction point. I need to invest time here. Will review everything else in between.